File: Runtime/InitialConditionsSpatial.vb
'*********************************************************************************************
' ST-Sim: A SyncroSim Module for the ST-Sim State-and-Transition Model.
'
' Copyright © 2007-2017 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.
'
'*********************************************************************************************

Friend Class InitialConditionsSpatial

    Private m_Iteration As Nullable(Of Integer)
    Private m_PrimaryStratumFileName As String
    Private m_SecondaryStratumFileName As String
    Private m_TertiaryStratumFileName As String
    Private m_StateClassFileName As String
    Private m_AgeFileName As String

    Public Sub New(
        ByVal iteration As Nullable(Of Integer),
        ByVal primaryStratumName As String,
        ByVal secondaryStratumFileName As String,
        ByVal tertiaryStratumFileName As String,
        ByVal stateClassFileName As String,
        ByVal ageFileName As String)

        Me.m_Iteration = iteration
        Me.m_PrimaryStratumFileName = primaryStratumName
        Me.m_SecondaryStratumFileName = secondaryStratumFileName
        Me.m_TertiaryStratumFileName = tertiaryStratumFileName
        Me.m_StateClassFileName = stateClassFileName
        Me.m_AgeFileName = ageFileName

    End Sub

    Public ReadOnly Property Iteration As Integer?
        Get
            Return Me.m_Iteration
        End Get
    End Property

    Public ReadOnly Property PrimaryStratumFileName As String
        Get
            Return Me.m_PrimaryStratumFileName
        End Get
    End Property

    Public ReadOnly Property SecondaryStratumFileName As String
        Get
            Return Me.m_SecondaryStratumFileName
        End Get
    End Property

    Public ReadOnly Property TertiaryStratumFileName As String
        Get
            Return Me.m_TertiaryStratumFileName
        End Get
    End Property

    Public ReadOnly Property StateClassFileName As String
        Get
            Return Me.m_StateClassFileName
        End Get
    End Property

    Public ReadOnly Property AgeFileName As String
        Get
            Return Me.m_AgeFileName
        End Get
    End Property
End Class

File: src/Runtime/InitialConditionsSpatial.cs
// A SyncroSim Package for developing state-and-transition simulation models using ST-Sim.
// Copyright © 2007-2019 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.

namespace SyncroSim.STSim
{
    internal class InitialConditionsSpatial
    {
        private int? m_Iteration;
        private string m_PrimaryStratumFileName;
        private string m_SecondaryStratumFileName;
        private string m_TertiaryStratumFileName;
        private string m_StateClassFileName;
        private string m_AgeFileName;

        public InitialConditionsSpatial(
            int? iteration, string primaryStratumName, string secondaryStratumFileName, 
            string tertiaryStratumFileName, string stateClassFileName, string ageFileName)
        {
            this.m_Iteration = iteration;
            this.m_PrimaryStratumFileName = primaryStratumName;
            this.m_SecondaryStratumFileName = secondaryStratumFileName;
            this.m_TertiaryStratumFileName = tertiaryStratumFileName;
            this.m_StateClassFileName = stateClassFileName;
            this.m_AgeFileName = ageFileName;
        }

        public int? Iteration
        {
            get
            {
                return this.m_Iteration;
            }
        }

        public string PrimaryStratumFileName
        {
            get
            {
                return this.m_PrimaryStratumFileName;
            }
        }

        public string SecondaryStratumFileName
        {
            get
            {
                return this.m_SecondaryStratumFileName;
            }
        }

        public string TertiaryStratumFileName
        {
            get
            {
                return this.m_TertiaryStratumFileName;
            }
        }

        public string StateClassFileName
        {
            get
            {
                return this.m_StateClassFileName;
            }
        }

        public string AgeFileName
        {
            get
            {
                return this.m_AgeFileName;
            }
        }
    }
}
